hypothesis-python/src/hypothesis/strategies/_internal/utils.py

```diff
@@ -167,24 +168,30 @@ def to_jsonable(obj: object) -> object:
         if isinstance(obj, int) and abs(obj) >= 2**63:
             # Silently clamp very large ints to max_float, to avoid
             # OverflowError when casting to float.
+            if avoid_realization:
+                # This adds another path for solvers to deal with, but (probably?)
+                # the ability to see small integers in tooling is worth it.
+                return "<symbolic>"
```

Comment:
It would be nice if solver-based backends could provide an `is_symbolic` function that we could check here, so we don't have to choose between replacing everything or nothing with <symbolic> 🤔 

(I think this check also doesn't match the comment, because this branch is only taken for `abs(obj) > 2**63`, not small ints)